Applications/AdminUi/apps/admin_ui/lib/core/widgets/quotas_button_group.dart

```diff
@@ -80,10 +82,29 @@ class _QuotasButtonGroupState extends State<QuotasButtonGroup> {
     }
   }
 
-  Future<ApiResponse<void>> _deleteQuota(String quota) {
+  Future<ApiResponse<void>> _deleteQuota(String quota) async {
     final client = GetIt.I.get<AdminApiClient>();
 
-    if (widget.identityAddress != null) return client.identities.deleteIndividualQuota(address: widget.identityAddress!, individualQuotaId: quota);
-    return client.quotas.deleteTierQuota(tierId: widget.tierId!, tierQuotaDefinitionId: quota);
+    unawaited(
+      showDialog<void>(
+        barrierDismissible: false,
+        builder: (ctx) {
+          return const Center(child: CircularProgressIndicator());
+        },
+        context: context,
+      ),
+    );
+
+    ApiResponse<void> response;
+
+    if (widget.identityAddress != null) {
+      response = await client.identities.deleteIndividualQuota(address: widget.identityAddress!, individualQuotaId: quota);
+    } else {
+      response = await client.quotas.deleteTierQuota(tierId: widget.tierId!, tierQuotaDefinitionId: quota);
+    }
```

Comment:
I'd be a fan of putting this in a separate method to not clutter this with a floating variable

Reply:
I restored the original method, because I had to move the dialog to the "parent" method anyway.